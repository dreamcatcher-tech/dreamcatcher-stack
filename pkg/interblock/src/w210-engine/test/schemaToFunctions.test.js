import { schemaToFunctions } from '..'
import Debug from 'debug'
const debug = Debug('interblock:tests:schemaToFunctions')

describe('schemaToFunctions', () => {
  test('basic', () => {
    const actions = {
      add: {
        type: 'object',
        title: 'ADD',
        description: 'Add an element to this collection',
        additionalProperties: false,
        properties: {
          test: { type: 'string' },
        },
      },
    }
    const functions = schemaToFunctions(actions)
    const add = functions.add()
    debug(add)
    expect(add).toMatchSnapshot()
  })
})
