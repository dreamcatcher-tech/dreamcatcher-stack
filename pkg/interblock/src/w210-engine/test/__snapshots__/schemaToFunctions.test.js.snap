// Vitest Snapshot v1

exports[`schemaToFunctions > basic 1`] = `
{
  "payload": {},
  "type": "ADD",
}
`;
